### PR TITLE
Changed the inital state of books

### PR DESCRIPTION
**Pull request by MohamedRoshdy2021**
in this PR i have 

- Added more initial books to the initial state 
- enhanced the functionality of the delete / edit methods 
- Added author input who takes the value from the user

### CODE REVIEW

File: src/components/Books.js [hunk 1]
```diff
@@ -113,6 +114,7 @@ function Books() {
         <h1>Add New Book</h1>
         <form onSubmit={handleAddBook}>
```
Comment (skyv26) [inline src/components/Books.js]: - [ ] Please 🙏🏻 adhere to the project requirements, according to which you should add the `onClick` event on the **Button** component, which will help to add the new book into the `redux` store using the reducer. Right now you are using the `onSubmit` event, which is fine but not fulfilling the requirement. You should create a Button.js file for creating the component and then import that component and add the `click` event to make the things working as it working right now.

### Requirement
Screenshot from 2023-08-23 21-53-57


### Caution ⚠️ 
**Please you need to replace your form submit button with the <Button onClick={Your_Event_Callback}> Add Book </Button>**.

File: src/components/Books.js
Comment (skyv26) [inline src/components/Books.js:1]: - [ ] Please 🙏🏻 adhere to the project requirements, according to which you should add the `onClick` event on the **Button** component, which will help to remove the specific book from the `redux` store using the reducer. Right now you are using the `button` which is JSX, not a **Component**, which is fine but not fulfilling the requirement. You should re-use the Button.js file for creating the component or Inside the Books.js file you can create one more component with Button and then import it to remove any book using the `click` event to make the things working as it working right now.

### Requirement
Screenshot from 2023-08-23 21-53-57

Comment (skyv26) [inline src/components/Books.js:1]: - [OPTIONAL] I would like to suggest you, please make more component out of your existing codebase, which will help you to debug the code easily and also will help someone to understand your code in more better way.

File: src/components/Books.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { addBook, removeBook, editBook } from '../redux/Books/booksSlice';
 function Books() {
   const dispatch = useDispatch();
   const books = useSelector((state) => state.books);
+  console.log(books);
```
Comment (skyv26) [inline src/components/Books.js]: - [OPTIONAL] Please avoid the use of `console` statements while pushing your code for review. Having the `console` statement actually impacts the performance of the application. You are using only one so there is not serious issue, but having multiple can cause serious issues while developing big application. You can use it while building application but don't leave statement on staging of your changes.

File: src/components/Books.js
Comment (skyv26) [inline src/components/Books.js:1]: - [ ] Please make sure that your application is error-free, I noticed that your application has an error related to **key** for your book *map* iteration statement. Kindly fix this issue and make sure that your application is running error before ask for code review. Having errors in the application expose your code to the public, so it is very important to double check your work on the console.

Screenshot from 2023-08-23 21-48-30

**Review by skyv26 — CHANGES_REQUESTED** [review]
## Hi @MohamedRoshdy2021, 

Good job so far 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## STATUS: REQUIRE CHANGES 🔴 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implement the features ✅ 
- Nice *UI* design ✅ 
- Professional `README` template ✅ 
- Linters are green ✅ 
- Correct `GitFlow` ✅ 
- Correct `.github/workflows` for linters ✅ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

#### Cheers, and Happy coding!👏👏👏

## ⚠️  WARNING ⚠️ 
**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Books.js
Comment (Strangeal) [reply src/components/Books.js:1]: - [ ] Great job implementing the changes as suggested by the previous reviewer. However, I noticed you still have the unique key prop `warning` in your console that the previous reviewer mentioned. Kindly `fix` this issue and make sure that your application is free of any `warnings` before resubmitting for review.

Happy coding!😊🚀🍀

**Review by Strangeal — CHANGES_REQUESTED** [review]
Hi @MohamedRoshdy2021, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Alt Text


 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by MohamedRoshdy2021** [comment]
> Hi @MohamedRoshdy2021,
> 
> Great work on making the changes requested by a previous reviewer 👏 You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
> 
> Alt Text [ Alt Text ](https://camo.githubusercontent.com/4f9601b422a77702aefe8e13485d2d5618cc0c30772eb2aa9f4129eac4757e06/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4b7942657366386b4e5436733838534567622f67697068792e676966) [ ](https://camo.githubusercontent.com/4f9601b422a77702aefe8e13485d2d5618cc0c30772eb2aa9f4129eac4757e06/68747470733a2f2f6d656469612e67697068792e636f6d2f6d656469612f4b7942657366386b4e5436733838534567622f67697068792e676966)
> 
> ## Required Changes ♻️
> _Check the comments under the review._
> 
> ## Optional suggestions
> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
> 
> Cheers and Happy coding!👏👏👏
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
> 
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — APPROVED** [review]
Hi @MohamedRoshdy2021 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
 
The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**